lib/rucio/core/replica.py

```diff
@@ -1993,7 +2082,8 @@ def __cleanup_after_replica_deletion(scope_name_temp_table, scope_name_temp_tabl
     # Remove Archive Constituents
     constituent_associations_to_delete = set()
     if affected_archives:
-        session.query(scope_name_temp_table).delete()
+        stmt = delete(scope_name_temp_table)
+        session.execute(stmt)
         session.execute(insert(scope_name_temp_table), [sn._asdict() for sn in affected_archives])
```

Comment:
https://github.com/rucio/rucio/pull/6591/files#r1549891959

Reply:
Changed to:

```
        values = [sn._asdict() for sn in affected_archives]
        stmt = insert(scope_name_temp_table)
        session.execute(stmt, values)
```